### DISCUSSION TITLE
T-swap tests- Error when importing forge-std

### DISCUSSION THREAD

**Discussion by SquilliamX** [Cyfrin/security-and-auditing-full-course-s23#154] [Q&A] [answered]
<img width="953" alt="image" src="https://github.com/Cyfrin/security-and-auditing-full-course-s23/assets/134108406/abba3416-779c-4113-89b7-932cc9f1fd45">

The image attached is what i see in my vscode. I am trying to create the Invariant.t.sol file to test the T-swap codebase but It is throwing an error whenever I try to import forge-std, why is this happening and can anyone help me?

the error i am receiving are:
`Source "forge-std/Test.sol" not found: File not found. Searched the following locations: "".solidity(6275)`

and

`Source "forge-std/invariant/StdInvariant.sol" not found: File not found. Searched the following locations: "".solidity(6275)`

why am i getting this error and how can i fix this?

**Comment by EngrPips** [comment]
Hello @SquilliamX , please run the below code in your command line

```javascript
forge install foundry-rs/forge-std --no-commit
```

**Reply by SquilliamX** [reply]
okay i ran that code in my command line but the error persists still :(

**Reply by alymurtazamemon** [reply]
@SquilliamX Run `forge build` command and see if it gives an error otherwise it is ok and you can ignore these red lines. Some solidity extensions have different folder structure settings and because of that, they don't recognise the import files correctly.

**Reply by SquilliamX** [reply]
when i run `forge build` this is the output:

`$ forge build`
`
[⠃] Compiling...2024-04-12T01:38:45.707344Z ERROR foundry_compilers::artifacts: error="/home/smartchain/code/security-course/5-t-swap-audit/lib/forge-std/src/invariant/StdInvariant.sol": No such file or directory (os error 2)
[⠊] Compiling...
Error: 
failed to resolve file: "/home/smartchain/code/security-course/5-t-swap-audit/lib/forge-std/src/invariant/StdInvariant.sol": No such file or directory (os error 2); check configured remappings
        --> /home/smartchain/code/security-course/5-t-swap-audit/test/unit/invariant/Invariant.t.sol
        forge-std/invariant/StdInvariant.sol`

**Reply by alymurtazamemon** [reply]
@SquilliamX import this;

```solidity
import {StdInvariant} from "forge-std/StdInvariant.sol";
```

**Reply by SquilliamX** [reply]
lol thanks this fixed my Std-Invariant import.

**Comment by SquilliamX** [comment, marked as answer]
Update: I ran  `forge install foundry-rs/forge-std --no-commit` and after a few minutes the error went away. Also my Std-Invariant was wrong and was fixed to `import {StdInvariant} from "forge-std/StdInvariant.sol";`
